p2p/trust/trustmetric.go

```diff
@@ -0,0 +1,394 @@
+package trust
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+var (
+	store *trustMetricStore
+)
+
+type peerMetricRequest struct {
+	Key  string
+	Resp chan *TrustMetric
+}
+
+type trustMetricStore struct {
+	PeerMetrics map[string]*TrustMetric
+	Requests    chan *peerMetricRequest
+	Disconn     chan string
+}
+
+func init() {
```

Comment:
lets try to model the trustMetricStore like the AddrBook, rather than just spawning it in the `init()`